gsa/src/web/pages/extras/trashcanpage.js

```diff
@@ -119,105 +114,81 @@ const separateByUsageType = inputList => {
-  constructor(...args) {
-    super(...args);
-    this.state = {
-      trash: undefined,
-      loading: false,
-    };
-
-    this.createContentRow = this.createContentRow.bind(this);
-    this.createContentsTable = this.createContentsTable.bind(this);
-    this.getTrash = this.getTrash.bind(this);
-    this.handleEmpty = this.handleEmpty.bind(this);
-    this.handleDelete = this.handleDelete.bind(this);
-    this.handleRestore = this.handleRestore.bind(this);
-    this.handleErrorClose = this.handleErrorClose.bind(this);
-  }
+const Trashcan = () => {
+  const gmp = useGmp();
+  const [, renewSessionTimeout] = useUserSessionTimeout();
 
-  componentDidMount() {
-    this.getTrash();
-  }
+  const [trash, setTrash] = useState({});
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState();
 
-  getTrash() {
-    const {gmp} = this.props;
+  const loadTrashCanData = useCallback(() => {
+    setLoading(true);
     const data = gmp.trashcan.get().then(
       response => {
-        const trash = response.data;
-        this.setState({trash});
+        const trash_response = response.data;
+        setTrash(trash_response);
+        setLoading(false);
       },
-      error => {
-        this.setState({error});
+      err => {
+        setError(err);
+        setLoading(false);
       },
     );
     return data;
-  }
+  }, [gmp.trashcan]);
 
-  handleInteraction() {
-    const {onInteraction} = this.props;
-    if (isDefined(onInteraction)) {
-      onInteraction();
-    }
-  }
+  useEffect(() => {
+    loadTrashCanData();
+  }, [loadTrashCanData]);
 
-  handleRestore(entity) {
-    const {gmp} = this.props;
+  const handleInteraction = useCallback(() => {
+    renewSessionTimeout();
+  }, [renewSessionTimeout]);
 
-    this.handleInteraction();
+  const handleRestore = useCallback(
+    entity => {
+      handleInteraction();
 
-    return gmp.trashcan
-      .restore(entity)
-      .then(this.getTrash)
-      .catch(error => {
-        this.setState({
-          error: error,
-        });
-      });
-  }
+      return gmp.trashcan
+        .restore(entity)
+        .then(loadTrashCanData)
+        .catch(setError);
+    },
+    [loadTrashCanData, gmp.trashcan, handleInteraction],
+  );
 
-  handleDelete(entity) {
-    const {gmp} = this.props;
+  const handleDelete = useCallback(
+    entity => {
+      handleInteraction();
 
-    this.handleInteraction();
+      return gmp.trashcan.delete(entity).then(loadTrashCanData).catch(setError);
+    },
+    [loadTrashCanData, gmp.trashcan, handleInteraction],
+  );
 
-    return gmp.trashcan
-      .delete(entity)
-      .then(this.getTrash)
-      .catch(error => {
-        this.setState({
-          error: error,
-        });
-      });
-  }
-
-  handleEmpty() {
-    const {gmp} = this.props;
-    this.handleInteraction();
+  const handleEmpty = useCallback(() => {
+    handleInteraction();
 
-    this.setState({loading: true});
+    setLoading(true);
```

Comment:
I still don't get this loading case. Why is loading set to true until the loading is actually started? Maybe just remove setting loading here in handleEmpty